ring/consul_client.go

```diff
@@ -206,6 +212,65 @@ func (c *consulClient) WatchPrefix(prefix string, factory func() interface{}, do
 	}
 }
 
+// WatchKey will watch a given key in consul for changes. When the value
+// under said key changes, the f callback is called with the deserialised
+// value. To construct the deserialised value, a factory function should be
+// supplied which generates an empty struct for WatchPrefix to deserialise
+// into. Values in Consul are assumed to be JSON. This function blocks until
+// the done channel is closed.
```

Comment:
No thats correct - see https://inconshreveable.com/07-08-2014/principles-of-designing-go-apis-with-channels/

As this channels is purely used for triggering stoppage, I think its fine - its pretty common pattern (see done channels in contexts or the cancel channels on http requests).

I will add a direction to it though.